shim/handler.go

```diff
@@ -20,10 +20,16 @@ const (
 	ready       state = "ready"       // ready for requests
 )
 
-// PeerChaincodeStream interface for stream between Peer and chaincode instance.
+// PeerChaincodeStream is the common stream interface for Peer - chaincode communication.
+// Both chaincode-as-server and chaincode-as-client patterns need to support this
 type PeerChaincodeStream interface {
 	Send(*pb.ChaincodeMessage) error
 	Recv() (*pb.ChaincodeMessage, error)
+}
+
+// ClientStream supports the (original) chaincode-as-client interaction pattern
```

Comment:
I realize the name is now ambiguous, but I'm a little concerned about the API breakage here.  Removing a method from an interface is an incompatible change.  I'm not sure it's a deal breaker, but it would be nice if we could avoid it.

Reply:
Uses use the Chaincode interface which does not expose this. The only other user is the peer side "inProcStream" which was supplying the ClosSend method because it was forced to ... it wasn't doing anything with it. I think its not a deal break (I have built and tested peer and chaincode server with these changes).